ams/server/src/main/java/com/netease/arctic/server/manager/FlinkOptimizerContainer.java

```diff
@@ -285,8 +285,8 @@ private <T> T fetchCommandOutput(Process exec, Function<String, T> commandReader
             break;
           }
           LOG.info("{}", readLine);
-          if (commandReader != null) {
-            value = commandReader.apply(readLine);
+          if ((value = commandReader.apply(readLine)) != null) {
```

Comment:
We should keep the null check on `commandReader` to avoid throwing NPE exceptions.

Reply:
There should be no NPE here.